x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/IndexResolver.java

```diff
@@ -143,8 +142,7 @@ public static IndexResolution mergedMappings(String indexPattern, FieldCapabilit
                 );
             fields.put(name, field);
 
-            var isPartiallyUnmapped = fcs.size() < numberOfIndices;
-            if (isPartiallyUnmapped) {
+            if (fieldCapsResponse.getIndexResponses().stream().allMatch(fcir -> fcir.get().containsKey(fullName)) == false) {
```

Comment:
I suspect this could be a bit expensive for many fields or many indices or both cases.
Lets check if this affects `esql/from_idx_limit_1` benchmark after merging.